src/admin/mod.rs

```diff
@@ -182,48 +120,76 @@ mod handlers {
-                    if let Some(description) = description {
-                        data.push(format!(
-                            "# TYPE {name} counter\n# HELP {name} {description}\n{name} {value}"
-                        ));
-                    } else {
-                        data.push(format!("# TYPE {name} counter\n{name} {value}"));
-                    }
+            let any = match metric.as_any() {
+                Some(any) => any,
+                None => {
+                    continue;
                 }
-                Metric::Gauge(name, description, value) => {
-                    if let Some(description) = description {
-                        data.push(format!(
-                            "# TYPE {name} gauge\n# HELP {name} {description}\n{name} {value}"
-                        ));
-                    } else {
-                        data.push(format!("# TYPE {name} gauge\n{name} {value}"));
-                    }
+            };
+
+            let name = metric.name();
+
+            if let Some(counter) = any.downcast_ref::<Counter>() {
+                let value = counter.value();
+                if let Some(description) = metric.description() {
+                    data.push(format!(
+                        "# TYPE {name} counter\n# HELP {name} {description}\n{name} {value}"
+                    ));
+                } else {
+                    data.push(format!("# TYPE {name} counter\n{name} {value}"));
+                }
+            } else if let Some(gauge) = any.downcast_ref::<Gauge>() {
+                let value = gauge.value();
+
+                if let Some(description) = metric.description() {
+                    data.push(format!(
+                        "# TYPE {name} gauge\n# HELP {name} {description}\n{name} {value}"
+                    ));
+                } else {
+                    data.push(format!("# TYPE {name} gauge\n{name} {value}"));
                 }
-                Metric::Percentiles(name, description, percentiles) => {
-                    for (_label, percentile, value) in percentiles {
-                        if let Some(value) = value {
-                            if let Some(description) = description {
-                                data.push(format!(
-                                    "# TYPE {name} gauge\n# HELP {name} {description}\n{name}{{percentile=\"{:02}\"}} {value}",
-                                    percentile,
-                                ));
-                            } else {
-                                data.push(format!(
-                                    "# TYPE {name} gauge\n{name}{{percentile=\"{:02}\"}} {value}",
-                                    percentile,
-                                ));
-                            }
+            } else if any.downcast_ref::<AtomicHistogram>().is_some() {
+                let key = if let Ok(h) = HistogramMetric::try_from(metric.name()) {
```

Comment:
Should we move 165:180 into its own function (something of the form `fn get_histogram_percentiles() -> Option<Vec<&'static str, f64, u64)>`. It can return `None` in the case that the metric name doesn't match or the `delta.get` fails; on success, it can return the collected vector.

That function can then be repeated across the three metric endpoints.